x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/ExpressionBuilder.java

```diff
@@ -314,7 +330,35 @@ public NamedExpression visitQualifiedNamePattern(EsqlBaseParser.QualifiedNamePat
                 objects.add(".");
             }
 
-            String patternContext = patterns.get(i).ID_PATTERN().getText();
+            String patternContext = "";
+            if (patterns.get(i).ID_PATTERN() != null) {
+                patternContext = patterns.get(i).ID_PATTERN().getText();
+            } else if (patterns.get(i).params() != null) {
+                Expression exp = expression(patterns.get(i).params());
+                if (exp instanceof Literal lit) {
+                    // only Literal.NULL can happen with missing params, params for constants are not allowed
+                    if (lit.value() != null) {
+                        throw new ParsingException(src, "Constant [{}] is unsupported for [{}]", patterns.get(i), ctx.getText());
+                    }
+                } else if (exp instanceof UnresolvedAttribute ua) {
+                    String identifier = ua.name();
+                    if (identifier != null) {
+                        if (identifier.contains(WILDCARD) && (identifier.charAt(0) == '`') == false) {
```

Comment:
this check looks incomplete - extract an utility method inside IdentifierBuilder (because that's where the methods for escaping are used) and call it here. 
Instead of using the raw string, consider always escaping the value since the type is UnresolvedAttribute.

Or even better modify the code below to incorporate these resolved types directly instead of exporting them as strings only for them to be imported again.

Reply:
If the parameter is an identifier(`UnresolvedAttribute`), it will be appended to the patternString(quoted), objects(unquoted) and nameString(unquoted), the code to process fragments will be skipped.